### ISSUE TITLE
Support self-hosted GitLab

### ISSUE THREAD

**Issue by rarkins** [renovatebot/pro#3]
This issue is for tracking GitLab support in Renovate Pro.

Configuration:
- We can determine the platform type based on whether `GITHUB_ENDPOINT` or `GITLAB_ENDPOINT` is present
- Will use a GITLAB_TOKEN

Scheduler:
- Will need to use the "getRepos" function, same as `--autodiscover`

Webhook listener:
- We need to work out a feasible way to configure webhooks for each repository
- Adding project hooks: https://docs.gitlab.com/ce/api/projects.html#add-project-hook

Also, rate limiting for a single token is of concern.

**Comment by oleblaesing** [comment]
It this possible with the free version? I can't find any information on it in the docs. :(

**Comment by rarkins** [comment]
@oleblaesing yes, the Renovate CLI tool and corresponding Docker image already support GitLab. Please see the updated self-hosted documentation here: https://github.com/renovatebot/renovate/blob/master/docs/self-hosting.md
